gpflow/models/training_mixins.py

```diff
@@ -0,0 +1,58 @@
+import abc
+import collections
+from typing import Callable, Optional, Tuple, TypeVar, Union
+
+
+import tensorflow as tf
+import numpy as np
+
+InputData = tf.Tensor
+OutputData = tf.Tensor
```

Comment:
what are we gaining with these types? As a user, I might be a bit bewildered by all these types which just turn out to all be aliases for `tf.Tensor`

Reply:
Make the models a bit more self-documenting in regards to whether you pass in an input or an output or something else? Note that they only got moved in this PR.